lib/DidKeyDriver.js

```diff
@@ -159,31 +179,48 @@ export class DidKeyDriver {
    * Converts an Ed25519KeyPair object to a `did:key` method DID Document.
    *
    * @param {object} options - Options hashmap.
-   * @param {LDKeyPair|object} options.keyPair - Key used to generate the DID
+   * @param {object} options.keyPair - Key used to generate the DID
    *   document (either an LDKeyPair instance containing public key material,
    *   or a "key description" plain object (such as that generated from a KMS)).
-   *
+   * @param {object} [options.keyAgreementKeyPair] -  Optional
+   *   keyAgreement key pair for generating did for keyAgreement.
    * @returns {Promise<{didDocument: object, keyPairs: Map}>}
    *   Resolves with the generated DID Document, along with the corresponding
    *   key pairs used to generate it (for storage in a KMS).
    */
-  async _keyPairToDidDocument({keyPair} = {}) {
-    const verificationKeyPair =
-      await this.verificationSuite.from({...keyPair});
+  async _keyPairToDidDocument({keyPair, keyAgreementKeyPair} = {}) {
+    let {publicKeyMultibase} = keyPair;
+    if(!publicKeyMultibase && keyPair.publicKeyBase58) {
+      publicKeyMultibase = keyPair.fingerprint();
```

Comment:
It may be possible for this to return a promise.

```suggestion
      // handle backwards compatibility w/older key pair interfaces
      publicKeyMultibase = await keyPair.fingerprint();
```